src/node-query.js

```diff
@@ -3,7 +3,16 @@ import createGid from './create-gid';
 
 export default function nodeQuery(type, fields) {
   return function(parentSelection, fieldName, id) {
-    parentSelection.add(fieldName, {args: {id: createGid(type, id)}}, (node) => {
+    const gidProtocol = 'gid:';
+    let gid;
+
+    if (String(id).substr(0, gidProtocol.length) === gidProtocol) {
+      gid = id;
+    } else {
+      gid = createGid(type, id);
+    }
```

Comment:
Maybe pull this into a `normailzeId` function to keep it encapsulated and portable?